include/secp256k1_ellswift.h

```diff
@@ -0,0 +1,176 @@
+ *  In:      ell64:      pointer to a 64-byte array to decode
+ *
+ * This function runs in variable time.
+ */
+SECP256K1_API int secp256k1_ellswift_decode(
+    const secp256k1_context *ctx,
+    secp256k1_pubkey *pubkey,
+    const unsigned char *ell64
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Compute an ElligatorSwift public key for a secret key.
+ *
+ *  Returns: 1: secret was valid, public key was stored.
+ *           0: secret was invalid, try again.
+ *  Args:    ctx:        pointer to a context object, initialized for signing.
+ *  Out:     ell64:      pointer to a 64-byte area to receive the ElligatorSwift
+ *                       public key
+ *  In:      seckey32:   pointer to a 32-byte secret key.
+ *           auxrand32:  (optional) pointer to 32 bytes of randomness
+ *
+ * Constant time in seckey and auxrand32, but not in the resulting public key.
+ *
+ * It is recommended that auxrand32 contains 32 uniformly random bytes, though
+ * it optional (and does result in encodings that are indistinguishable from
+ * uniform without any auxrand32). It differs from the (mandatory) rnd32
+ * argument to secp256k1_ellswift_encode in this regard.
+
+ * This function can be used instead of calling secp256k1_ec_pubkey_create
+ * followed by secp256k1_ellswift_encode. It is safer, as it uses the secret
+ * key as entropy for the encoding (supplemented with auxrand32, if provided).
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_ellswift_create(
+    const secp256k1_context *ctx,
+    unsigned char *ell64,
+    const unsigned char *seckey32,
+    const unsigned char *auxrand32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Given a private key, and ElligatorSwift public keys sent in both directions,
+ *  compute a shared secret using x-only Elliptic Curve Diffie-Hellman (ECDH).
+ *
+ *  Returns: 1: shared secret was succesfully computed
+ *           0: secret was invalid or hashfp returned 0
+ *  Args:    ctx:       pointer to a context object.
+ *  Out:     output:    pointer to an array to be filled by hashfp.
+ *  In:      theirs64:  a pointer to the 64-byte ElligatorSwift public key
+ *                      received from the other party.
+ *           ours64:    a pointer to the 64-byte ElligatorSwift public key
+ *                      sent to the other party.
+ *           seckey32:  a pointer to the 32-byte secret key corresponding to
+ *                      ours64 (the correspondence is not checked)
+ *           hashfp:    pointer to a hash function. If NULL,
+ *                      secp256k1_ellswift_xdh_hash_function_default is used
+ *                      (in which case, 32 bytes will be written to output).
+ *           data:      arbitrary data pointer passed through to hashfp (ignored
+ *                      by secp256k1_ellswift_xdh_hash_function_default).
```

Comment:
Seems like `secp256k1_ellswift_xdh_hash_function_default` doesn't exist anymore:
```suggestion
 *           data:      arbitrary data pointer passed through to hashfp.
```